File: exams/1617/translate/argparse/student/java/StringParameter.java
import java.util.ArrayList;


class StringParameter extends Parameter
{
    private String value;

    public StringParameter(String name, String defaultValue)
    {
        super(name);

        this.value = defaultValue;
    }

    public boolean parse(ArrayList<String> args)
    {
        if ( args.size() >= 2 && args.get(0).equals("--" + name) )
        {
            this.value = args.get(1);
            args.remove(0);
            args.remove(0);

            return true;
        }
        else
        {
            return false;
        }
    }

    public String value()
    {
        return value;
    }

    // Te vertalen als globale functie, d.i. een functie buiten de klasse
    public static StringParameter string(String name, String defaultValue)
    {
        return new StringParameter(name, defaultValue);
    }
}

File: exams/1617/translate/argparse/solution/argparse/string-parameter.cpp
#include "string-parameter.h"


StringParameter::StringParameter(const std::string& name, const std::string& default_value)
	: Parameter(name), m_value(default_value) { }

bool StringParameter::parse(std::list<std::string>& args)
{
	if (!args.size() >= 2 && args.front() == "--" + m_name)
	{
		args.pop_front();
		m_value = args.front();
		args.pop_front();

		return true;
	}
	else
	{
		return false;
	}
}

std::string StringParameter::value() const
{
	return m_value;
}

std::shared_ptr<StringParameter> string(const std::string& name, const std::string& default_value)
{
	return std::make_shared<StringParameter>(name, default_value);
}
